lib/ohloh_scm/adapters/git_svn/pull.rb

```diff
@@ -0,0 +1,83 @@
+module OhlohScm::Adapters
+  class GitSvnAdapter < AbstractAdapter
+    def pull(source_scm, &block)
+      @source_scm = source_scm
+      convert_to_git(&block)
+    end
+
+    def branch_name
+      'master'
+    end
+
+    private
+
+    def convert_to_git(&block)
+      yield(1, 2) if block_given?
+
+      if FileTest.exist?(git_path)
+        max_step = @source_scm.commit_count(after: head_token)
+        fetch(max_step, &block)
+      else
+        max_step = @source_scm.commit_count(after: 0)
+        clone(max_step, &block)
+      end
+
+      clean_up_disk
+      yield(max_step, max_step) if block_given?
```

Comment:
We don't need to do this here. In the `IO.popen` loop, we start the count by 1, so it would yield correct values at the end. If we remove this, we can move the max_step lines into `fetch` and `clone` methods.